tests/test_invenio_previewer_views.py

```diff
@@ -28,147 +28,182 @@
-            with open(filename_path, 'w') as file:
-                file.write(content)
-
-            rec_uuid = uuid.uuid4()
-            with db.session.begin_nested():
-                Record.create({
-                    "title": "TestDefault",
-                    "files": [{"uri": filename_path}]
-                }, id_=rec_uuid)
-            url = url_for('invenio_previewer.document',
-                          recid=rec_uuid, filename=filename)
-            response = client.get(url)
-            assert content == response.data.decode('ascii')
-            shutil.rmtree(tmpdirname)
+    InvenioDB(app_assets)
+    InvenioFilesREST(app_assets)
+    with app_assets.test_request_context():
+        db.create_all()
+        tmpdirname = tempfile.mktemp()
+        if not os.path.exists(tmpdirname):
+            os.makedirs(tmpdirname)
+        file_name = 'zipfile.zip'
+        filename_path = os.path.join(tmpdirname, file_name)
+        zipf = zipfile.ZipFile(filename_path, 'w')
+        zipf.writestr('Example.txt', 'This is an example')
+        zipf.close()
+
+        pid_value = create_rest_files(tmpdirname, file_name, filename_path)
+        db.session.commit()
+
+        url = url_for('invenio_previewer.preview', pid=pid_value)
+        response = get_response_application(app_assets, url)
+        assert 'Example.txt' in response.content.decode('utf-8')
+        shutil.rmtree(tmpdirname)
+        db.drop_all()
+
+
+def create_rest_files(tmpdirname, file_name, filename_path):
+    rec_uuid = uuid.uuid4()
+    provider = RecordIdProvider.create(object_type='rec',
+                                       object_uuid=rec_uuid)
+    pid_value = provider.pid.pid_value
+    with db.session.begin_nested():
+        loc = Location(name='local', uri=tmpdirname, default=True)
+    with db.session.begin_nested():
+        bucket = Bucket.create(loc)
+        with open(filename_path, 'r') as file:
+            ObjectVersion.create(bucket, file_name, stream=file)
+        Record.create({
+            "pid_value": pid_value,
+            "title": "TestDefault",
+            "files": [
+                {
+                    "uri": '/files/{0}/{1}'.format(str(bucket.id),
+                                                   file_name),
+                    "name": file_name
+                }
+            ]
+        }, id_=rec_uuid)
+    return pid_value
+
+
+def get_response_application(application, url):
+    """Executes a threaded server and executes a real request."""
+    url = urljoin('http://localhost:5000/', url)
+
+    def run_server():
+        application.run(threaded=True)
+    thread = Process(target=run_server)
+    thread.start()
+    while True:
+        try:
+            response = requests.get(url, stream=True, timeout=0.5)
+            thread.terminate()
+            thread.join()
+            return response
+        except requests.exceptions.ConnectionError:
+            pass
```

Comment:
Although I am planing to use a counter in the ``while True`` loop, the method will still looking a bit ugly. I haven't found a better solution to solve the issue of nested HTTP requests. Is this solution ok?